File: src/api/Cryoptix.Strategy/State/StrategyStatus.cs
using Cryoptix.Strategy.Processor;

namespace Cryoptix.Strategy.State
{
    /// <summary>
    /// Represents the strategy status.
    /// </summary>
    public sealed class StrategyStatus
    {
        /// <summary>
        /// Gets or sets the strategy state.
        /// </summary>
        public StrategyState StrategyState { get; set; }
        /// <summary>
        /// Gets or sets the strategy processor type.
        /// </summary>
        public StrategyProcessorType StrategyProcessorType { get; set; }
        /// <summary>
        /// Gets or sets the strategy.
        /// </summary>
        public Strategies.Strategy? Strategy { get; set; }
        /// <summary>
        /// Gets or sets the message.
        /// </summary>
        public string? Message { get; set; }
    }
}

File: src/gui/cryoptix/src/features/strategy/strategy-status.ts
import { z } from "zod";
import { StrategySchema } from "./strategy-schema";
import { StrategyStateSchema } from "./strategy-state";
import { StrategyProcessorTypeSchema } from "./strategy-processor-type";

export const StrategyStatusApiSchema = z.object({
  StrategyState: StrategyStateSchema,
  StrategyProcessorType: StrategyProcessorTypeSchema,
  Strategy: StrategySchema.nullable(),
  Message: z.string().nullable(),
});

export const StrategyStatusSchema = StrategyStatusApiSchema.transform(
  (data) => ({
    strategyState: data.StrategyState,
    strategyProcessorType: data.StrategyProcessorType,
    strategy: data.Strategy,
    message: data.Message,
  })
);

export type StrategyStatus = z.infer<typeof StrategyStatusSchema>;
